src/WP_CLI/CommandWithMeta.php

```diff
@@ -254,6 +254,140 @@ public function update( $args, $assoc_args ) {
 	}
 
 	/**
+	 * Get a nested value from a meta field.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <id>
+	 * : The ID of the object.
+	 *
+	 * <key>
+	 * : The name of the meta field to get.
+	 *
+	 * <key-path>...
+	 * : The name(s) of the keys within the value to locate the value to pluck.
+	 *
+	 * [--format=<format>]
+	 * : The output format of the value.
+	 * ---
+	 * default: plaintext
+	 * options:
+	 *   - plaintext
+	 *   - json
+	 *   - yaml
+	 */
+	public function pluck( $args, $assoc_args ) {
+		list( $object_id, $meta_key ) = $args;
+		$object_id = $this->check_object_id( $object_id );
+		$key_path = array_map( function( $key ) {
+			if ( is_numeric( $key ) && ( $key === (string) intval( $key ) ) ) {
+				return (int) $key;
+			}
+			return $key;
+		}, array_slice( $args, 2 ) );
+
+		$value = get_metadata( $this->meta_type, $object_id, $meta_key, true );
+
+		$traverser = new RecursiveDataStructureTraverser( $value );
+
+		try {
+			$value = $traverser->get( $key_path );
+		} catch ( \Exception $e ) {
+			die( 1 );
```

Comment:
> Instead of silently dying here, you should use something like the following:

I looked into this. The existing behavior is how `wp option get` operates:

```
$ wp option get fooos
$ echo $?
1
```
I think we should mirror existing behavior as much as posssible.